config/initializers/hydra_config.rb

```diff
@@ -58,3 +58,19 @@ def byname
 
 # Clear the role map out of the Rails cache so it initializes from the DB
 Rails.cache.delete("RoleMapHash")
+
+ActiveFedora::QueryMethods.module_eval do
+  extend ActiveSupport::Concern
+
+  def accessible_by(ability, action = :index)
+    permission_types = case action
+      when :index then [:discover, :read, :edit]
+      when :show then [:read, :edit]
+      when :update, :edit, :create, :new, :destroy then [:edit]
+    end
+
+    builder = ::SearchBuilder.new(nil).with_ability(ability)
+    filters = builder.send(:gated_discovery_filters).reject(&:blank?).join(' OR ')
```

Comment:
I did some testing with this and having `.with_discovery_permissions(permission_types)` included caused every user to be given every policy id, so `accessible_by` would return all items regardless of actual permissions. Removing that method assigned the proper policy ids, but the admin user was still not being returned everything for some reason. So I added a ternary conditional in the media_object_controller to return `MediaObject.all` if the user is an admin.

I'm unclear why the admin's policy id didn't return everything. Here's the solr query generated from `filters`: 
```
get policies: [{"id"=>"v979v304g"}]


get policies: []


"_query_:\"{!terms f=edit_access_group_ssim}public,administrator,manager,group_manager,registered,172.18.0.1\" 
OR _query_:\"{!terms f=discover_access_group_ssim}public,administrator,manager,group_manager,registered,172.18.0.1\" 
OR _query_:\"{!terms f=read_access_group_ssim}public,administrator,manager,group_manager,registered,172.18.0.1\" 
OR edit_access_person_ssim:archivist1@example.com 
OR discover_access_person_ssim:archivist1@example.com
 OR read_access_person_ssim:archivist1@example.com 
OR (_query_:\"{!raw f=isGovernedBy_ssim}v979v304g\")"
```